mlflow/transformers/__init__.py

```diff
@@ -1333,6 +1333,22 @@ def _extract_torch_dtype_if_set(pipeline):
     if torch_dtype := getattr(pipeline, _TORCH_DTYPE_KEY, None):
         return str(torch_dtype)
 
+    # Transformers pipeline doesn't inherit underlying model's dtype, so we have to extract it
+    # from the model.
+    model = pipeline.model
+    if torch_dtype := getattr(model.config, _TORCH_DTYPE_KEY, None):
+        return str(torch_dtype)
```

Comment:
Seems like the model.config storage stores the string name of the underlying type instead of the object reference (i.e. "float16" vs. the expected `torch.float16`). 
While adding the formatting in accordance with the expected syntax might be expedient, we might want to validate that the attribute we're extracting is a valid torch dtype value during save and then store the string in the parseable format to ensure that it's resolvable once hitting `_torch_dtype_mapping()`

Reply:
Yes, I'll first check is there any convenient API in Transformer to do similar validation and conversion.

Reply:
Unfortunately Transformer doesn't provide convenient way to validate it, so added a custom impl (inside deser function) and run deser->ser to standardize the format.

Reply:
I figured that transformers wouldn't get involved, since it's entirely a torch feature :)